Source/DafnyLanguageServer/Plugins/QuickFixerHelpers.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using Microsoft.Boogie;
+using Microsoft.Dafny.LanguageServer.Language;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.Plugins;
+
+/// <summary>
+/// Helpers for plugins defining a QuickFixer
+/// </summary>
+public static class QuickFixerHelpers {
+  /// <summary>
+  /// Given a LSP range and some text, extract the corresponding substring
+  /// </summary>
+  /// <param name="range">The range to extract</param>
+  /// <param name="text">The document</param>
+  /// <returns>The substring of the document in the range</returns>
+  public static string Extract(Range range, string text) {
+    var token = range.ToBoogieToken(text);
+    var startTokenPos = token.StartToken.pos;
+    var endTokenPos = token.EndToken.pos + token.EndToken.val.Length;
+    var length = endTokenPos - startTokenPos;
+    if (startTokenPos < 0 || endTokenPos < startTokenPos || endTokenPos >= text.Length) {
+      return ""; // Safeguard
+    }
+
+    return text.Substring(startTokenPos, length);
+  }
+
+  /// <summary>
+  /// Given the position of a closing brace '}' and the position of the opening brace '{',
```

Comment:
Since matching-parentheses functionality is provided in the method below, maybe one can provide only the opening parenthesis here?

Reply:
Could you add a brief example showing what indentation this method would return for a simple code snippet?

Reply:
I added a brief example.
I prefer to dissociate provide both, because in any case, the caller will need the end token to perform the insertion, so it has to compute it.

However, I created another helper that do both at the same time so that's it's easier to use.